flink-runtime/src/main/java/org/apache/flink/streaming/api/graph/StreamGraphContext.java

```diff
@@ -62,4 +62,13 @@ public interface StreamGraphContext {
      * @return true if all modifications were successful and applied atomically, false otherwise.
      */
     boolean modifyStreamEdge(List<StreamEdgeUpdateRequestInfo> requestInfos);
+
+    interface StreamGraphUpdateListener {
+        /**
+         * This method is called whenever the StreamGraph is updated.
+         *
+         * @param streamGraph the updated StreamGraph
+         */
+        void onStreamGraphUpdated(StreamGraph streamGraph);
```

Comment:
I think the listener already holds a stream graph, so we don't need to pass a stream graph.

Reply:
It's true that `AdaptiveGraphManager` hold a stream graph, but the `StreamGraphContext` should notify it's listener with the updated stream graph (since the stream graph in the `AdaptiveGraphManager` maybe outdated?).
Just like `JobStatusListener` should notify listener with `newJobStatus`.

Reply:
The context of the stream graph is generated from the AdaptiveGraphManager. Therefore, it is natural for the context to hold the same stream graph as the one held by the AdaptiveGraphManager. On the other hand, it is strange that the listener does not know which stream graph it is listening to. Therefore, I think there is no need to pass a stream graph.

Reply:
And I checked the JobStatusListener you mentioned once again. You can see that the only job-specific element in its interface is the JobID, and in its implementation, this attribute is not used at all. In other words, although its interface may be designed this way for certain reasons, the job-specific identifier is completely redundant.

Reply:
As discussed offline with @JunRuiLee , the AdaptiveGraphManager has complete informations of the StreamGraph, we only need to notify the listener that the StreamGraph has been changed, so I removed the streamGraph parameter in the revised version.